include/ccf/odata_error.h

```diff
@@ -7,14 +7,28 @@
 
 namespace ccf
 {
+  struct ODataErrorDetails
+  {
+    std::string auth_policy;
+    std::string code;
+    std::string message;
+
+    bool operator==(const ODataErrorDetails&) const = default;
+  };
+
+  DECLARE_JSON_TYPE(ODataErrorDetails);
+  DECLARE_JSON_REQUIRED_FIELDS(ODataErrorDetails, auth_policy, code, message);
+
   struct ODataError
   {
     std::string code;
     std::string message;
+    std::vector<ODataErrorDetails> moredetails = {};
```

Comment:
This field should be called `details`, rather than `moredetails`, to match the field defined by the [OData spec](http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#_Toc38457792). We can insert any other fields we want (as we do with the `auth_policy` in `ODataErrorDetails`), but in this case we're implementing a standard/expected field.

Reply:
Done